In [1]:
import matplotlib.pyplot as plt
%matplotlib inline

import pandas as pd
import numpy as np
import seaborn as sns

from sklearn.model_selection import train_test_split
from sklearn.linear_model import LinearRegression
from sklearn import metrics


data = pd.read_csv('data/beer_consumption.csv', sep = ';')

# <font color = 'red' style = 'font-size: 30px;'> Saving and Loading the Estimated Model </font>
<hr style = 'border: 2px solid red;'>

## Input data

In [7]:
X = data[['max_temp', 'rain', 'weekend']]
y = data['consumption']

X_train, X_test, y_train, y_test = train_test_split(X, y, test_size= 0.3, random_state= 2811)

model = LinearRegression()
model.fit(X_train, y_train)

X_test[0:1]

max_temp  rain  weekend
5      30.5  12.2        0

In [5]:
input = X_test[0:1]

## Generating punctual forecast

In [8]:
model.predict(input)[0]

26094.901775262017

## Creating a simple simulator

In [9]:
max_temp = 30.5
rain=12.2
weekend=0
input=[[max_temp, rain, weekend]]

print('{0:.2f} liters'.format(model.predict(input)[0]))

26094.90 liters


## Saving the estimated model

## pickle (https://docs.python.org/3/library/pickle.html)

<p style = 'font-size: 20px; line-height: 2; margin: 10px 50px; text-align: justify; '> The <b> pickle </b> module implements binary protocols to serialize and deserialize the structure of a Python object. </p>

In [11]:
import pickle

In [12]:
output = open('beer_consumption_model', 'wb')
pickle.dump(model, output)
output.close()

### On a new Python notebook / project

<h4 style = 'color: blue; font-weight: normal '> In [1]: </h4>

`` sh
import pickle

modelo = open ('modelo_consumo_cerveja', 'rb')
lm_new = pickle.load (model)
model.close ()

temp_max = 30.5
rain = 12.2
fds = 0
entry = [[temp_max, rain, weekend]]
print ('{0: .2f} liters'.format (lm_new.predict (entry) [0]))
``

<h4 style = 'color: red; font-weight: normal '> Out [1]: </h4>

``
26094.90 liters
``

In [14]:
import pickle

imported_model = open('beer_consumption_model', 'rb')
lm_new = pickle.load(imported_model)
imported_model.close()

max_temp = 30.5
rain = 12.2
weekend = 0
entry = [[max_temp, rain, weekend]]
print ('{0: .2f} liters'.format (lm_new.predict (entry) [0]))

 26094.90 liters
